stdlib/public/Darwin/Accelerate/vImage_CVImageFormat.swift

```diff
@@ -0,0 +1,659 @@
+        set {
+            let error = vImageCVImageFormat_SetAlphaHint(self,
+                                                         newValue ? 2 : 0)
+            
+            if error != kvImageNoError {
+                fatalError("Unable to set `alphaIsOpaqueHint`")
+            }
+        }
+    }
+    
+    /// The number of channels, including alpha, for the Core Video image format.
+    public var channelCount: UInt32  {
+        return vImageCVImageFormat_GetChannelCount(cvConstImageFormat)
+    }
+    
+    /// The names of the channels of a Core Video image format.
+    public var channels: [vImage.BufferType] {
+        let channels = Array(UnsafeBufferPointer(start: vImageCVImageFormat_GetChannelNames(cvConstImageFormat),
+                                                 count: Int(channelCount)))
+        
+        return channels.compactMap {
+            return vImage.BufferType(bufferTypeCode: Int($0),
+                                     model: colorSpace?.model)
+        }
+    }
+    
+    /// Returns the channel description for a particular channel type.
+    public func channelDescription(bufferType: vImage.BufferType) -> vImageChannelDescription? {
+        guard let description = vImageCVImageFormat_GetChannelDescription(cvConstImageFormat,
+                                                                          bufferType.bufferTypeCode) else {
+                                                                            return nil
+        }
+        
+        return vImageChannelDescription(min: description.pointee.min,
+                                        zero: description.pointee.zero,
+                                        full: description.pointee.full,
+                                        max:  description.pointee.max)
+    }
+    
+    /// The chroma siting of a Core Video image format.
+    public var chromaSiting: ChromaSiting?  {
+        get {
+            return ChromaSiting(location: vImageCVImageFormat_GetChromaSiting(cvConstImageFormat)?.takeRetainedValue())
+        }
+        set {
+            let error = vImageCVImageFormat_SetChromaSiting(self,
+                                                            newValue?.cfString)
+            
+            if error != kvImageNoError {
+                fatalError("Unable to set `chromaSiting`")
+            }
+        }
+    }
+    
+    ///  The color space of a Core Video image format.
+    public var colorSpace: CGColorSpace?  {
+        get {
+            return vImageCVImageFormat_GetColorSpace(cvConstImageFormat)?.takeRetainedValue()
+        }
+        set {
+            let error = vImageCVImageFormat_SetColorSpace(self,
+                                                          newValue)
+            
+            if error != kvImageNoError {
+                fatalError("Unable to set `colorSpace`")
+            }
+        }
+    }
+    
+    /// The format code of a Core Video image format.
+    public var formatCode: String {
```

Comment:
I think these format codes would be better represented by a dedicated wrapper type around an `UInt32` value. Conversion between `String` and the type code representation is laborious.

When we do convert these to strings, we should make sure we're creating a small (inline) Swift string, in a consistent and efficient way. `UTCreateStringForOSType` incurs allocation, transcoding and bridging overhead, and its results are inconsistent with the values generated on non-macOS platforms (it uses Mac OS Roman, not Latin-1).


```
@_fixed_layout
public struct vImageFormatCode: RawRepresentable {
  @usableFromInline
  internal let rawValue: UInt32

  @inlinable
  public init(rawValue: UInt32)

  public var stringRepresentation: String {
    // FIXME: This uses Latin-1, not MacRoman.
    var result = ""
    result.append(String(UnicodeScalar(formatCode >> 24 & 0xFF)!))
    result.append(String(UnicodeScalar(formatCode >> 16 & 0xFF)!))
    result.append(String(UnicodeScalar(formatCode >> 8 & 0xFF)!))
    result.append(String(UnicodeScalar(formatCode >> 0 & 0xFF)!))
    return result
  }
}

extension vImageCVImageFormat {
  public formatCode: vImageFormatCode {
    let formatCode = vImageCVImageFormat_GetFormatCode(cvConstImageFormat)
    return vImageFormatCode(formatCode)
  }
}
```

Reply:
See above, this property is now a `UInt32`.